Python 3.5 is failing on Windows

This was passing on staged-recipes but failed here :unamused: 

```shell
LINK : fatal error LNK1181: cannot open input file 'Files.obj'
error: Command "C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:C:\conda\envs\_build\libs /LIBPATH:C:\conda\envs\_build\PCbuild\amd64 /LIBPATH:C:\conda\envs\_build\Library\lib /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\LIB\amd64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\LIB\amd64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.14393.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.14393.0\um\x64 /EXPORT:PyInit_interpreter build\temp.win-amd64-3.5\Release\numexpr\win32\pthread.obj build\temp.win-amd64-3.5\Release\numexpr\interpreter.obj build\temp.win-amd64-3.5\Release\numexpr\module.obj build\temp.win-amd64-3.5\Release\numexpr\numexpr_object.obj /OUT:build\lib.win-amd64-3.5\numexpr\interpreter.cp35-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.5\Release\numexpr\win32\interpreter.cp35-win_amd64.lib" failed with exit status 1181
Command failed: C:\windows\system32\cmd.exe /c bld.bat
Command exited with code 1
```

I will investigate this tomorrow but I appreciate any pointer from our Windows experts (@msarahan, @patricksnape, and @gillins :grimacing:)

Full log at: 
https://ci.appveyor.com/project/conda-forge/numexpr-feedstock/branch/master/job/wqma7l2ut58hd85s

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -16,7 +16,7 @@ build:
 requirements:
     build:
         - python
-        - setuptools
+        - setuptools <=23
         - numpy x.x
     run:
         - python
@@ -35,7 +35,7 @@ about:
     home: https://github.com/pydata/numexpr
     license: MIT
     license_file: LICENSE.txt
-    summary: Fast numerical expression evaluator for NumPy
+    summary: 'Fast numerical expression evaluator for NumPy.'
     description: |
       Numexpr is a fast numerical expression evaluator for NumPy. With it,
       expressions that operate on arrays (like "3*a+4*b") are accelerated and use

